app/src/main/java/com/RaphaelAGN/chucknorrisapp/MainActivity.kt

```diff
@@ -2,53 +2,56 @@ package com.RaphaelAGN.chucknorrisapp
 
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
-import android.widget.Button
-import android.widget.TextView
-import android.widget.Toast
-import com.RaphaelAGN.chucknorrisapp.endpoint.Endpoint
-import com.RaphaelAGN.chucknorrisapp.data.JokeModel
-import com.RaphaelAGN.chucknorrisapp.retrofitClient.RetrofitClient
-import retrofit2.Call
-import retrofit2.Callback
-import retrofit2.Response
+import android.util.Log
+import androidx.fragment.app.*
+import com.RaphaelAGN.chucknorrisapp.fragments.MainFragment
 
 class MainActivity : AppCompatActivity() {
 
-    var jokeButton: Button? = null
-    var jokeTextView: TextView? = null
+    val manager: FragmentManager = supportFragmentManager
+    val transaction: FragmentTransaction = manager.beginTransaction()
+    val TAG = "chuck_norris_app"
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        Log.i(TAG, "onCreate")
         setContentView(R.layout.activity_main)
 
-        jokeTextView = findViewById(R.id.joke_text_view)
-
-        jokeTextView?.text = "Click the button below to get a joke"
+        if(savedInstanceState == null) {
+            val fragment = MainFragment()
+            transaction.replace(R.id.main_fragment, fragment)
```

Comment:
You can call `supportFragmentManager.beginTransaction()` directly instead of define it globally.
e.g.:
```
supportFragmentManager
  .beginTransaction()
  .replace(R.id.main_fragment, fragment)
  .commit()
```